File: Implement_Stack_using_Queues_two.py
class MyStack:

    def __init__(self):
        self.q1 = deque()
        self.q2 = deque()

    def push(self, x: int) -> None:
        self.q1.append(x)

    def pop(self) -> int:
        while len(self.q1) > 1:
            self.q2.append(self.q1.popleft())

        popped_val = self.q1.popleft()
        self.q1, self.q2 = self.q2, self.q1

        return popped_val
        
    def top(self) -> int:
        while len(self.q1) > 1:
            self.q2.append(self.q1.popleft())

        top_val = self.q1[0]
        self.q2.append(self.q1.popleft())
        self.q1, self.q2 = self.q2, self.q1

        return top_val

    def empty(self) -> bool:
        return len(self.q1) == 0

File: Implement_Stack_using_Queues_two.js
class MyStack {
    constructor() {
        this.q1 = [];
        this.q2 = [];
    }

    push(x) {
        this.q1.push(x);
    }

    pop() {
        while (this.q1.length > 1) {
            this.q2.push(this.q1.shift());
        }

        const poppedVal = this.q1.shift();
        [this.q1, this.q2] = [this.q2, this.q1];

        return poppedVal;
    }

    top() {
        while (this.q1.length > 1) {
            this.q2.push(this.q1.shift());
        }

        const topVal = this.q1[0];
        this.q2.push(this.q1.shift());
        [this.q1, this.q2] = [this.q2, this.q1];

        return topVal;
    }

    empty() {
        return this.q1.length === 0;
    }
}
